google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java

```diff
@@ -536,21 +535,41 @@ public void onError(SpannerException e, boolean withBeginTransaction) {
                 SpannerExceptionFactory.createAbortedExceptionWithRetryDelay(
                     "Aborted due to failed initial statement", e, 0, 1)));
       }
+      SpannerException exceptionToThrow;
+      if (withBeginTransaction
+          && e.getErrorCode() == ErrorCode.CANCELLED
+          && e.getMessage().contains("invalidated by a later transaction")) {
```

Comment:
nit: could we extract a constant for this?

Reply:
Done.